fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/postgresql/PostgreSqlAdapter.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * (C) Copyright IBM Corp. 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.database.utils.postgresql;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Supplier;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.ibm.fhir.database.utils.api.IConnectionProvider;
+import com.ibm.fhir.database.utils.api.IDatabaseStatement;
+import com.ibm.fhir.database.utils.api.IDatabaseTarget;
+import com.ibm.fhir.database.utils.common.AddForeignKeyConstraint;
+import com.ibm.fhir.database.utils.common.CommonDatabaseAdapter;
+import com.ibm.fhir.database.utils.common.DataDefinitionUtil;
+import com.ibm.fhir.database.utils.model.ColumnBase;
+import com.ibm.fhir.database.utils.model.ForeignKeyConstraint;
+import com.ibm.fhir.database.utils.model.IdentityDef;
+import com.ibm.fhir.database.utils.model.PrimaryKeyDef;
+import com.ibm.fhir.database.utils.model.Table;
+
+/**
+ * A PostgreSql database target
+ */
+public class PostgreSqlAdapter extends CommonDatabaseAdapter {
+    private static final Logger logger = Logger.getLogger(PostgreSqlAdapter.class.getName());
+
+    // Different warning messages we track so that we only have to report them once
+    private enum MessageKey {
+        MULTITENANCY, CREATE_VAR, CREATE_PERM, ENABLE_ROW_ACCESS, DISABLE_ROW_ACCESS, PARTITIONING,
```

Comment:
best to have each entry on a single line

Reply:
done